feature/gnmi/telemetry_npu_asic_counters_test/tests/README.md

```diff
@@ -0,0 +1,176 @@
+        * 30*pps = in-pkts and out-pkts
+        * 30*pps*300 = in-bytes and out-bytes
+    * Stop traffic.
+
+*   Packet Counters - Host-Block
+
+    *   Subscribe to the following, and capture the values of the 4 leaves(1),
+        * /packet/host-interface-block/in-pkts
+        * /packet/host-interface-block/out-pkts
+        * /packet/host-interface-block/in-bytes (TBD)
+        * /packet/host-interface-block/in-bytes (TBD)
+    *   Define variance = 0.99
+    *   Send ping packets to ATEPort2IP with count 1000. Wait for 20 secs.
+    *   Capture the in-pkts and out-pkts leaves(2).
+    *   Calculation of step 2 - step 1 leaf values.
+        * Step 2 - Step 1 >= 1000
+        * Step 2/Step 1 >= variance
+    *   Verify in-bytes and out-bytes (TBD)
+
+*   Packet Countes - Queueing Block
+
+    *   Subscribe to the following, and capture the values of the 4 leaves(1),
+        * /packet/queueing-block/in-pkts(TBD)
+        * /packet/queueing-block/in-bytes(TBD)
+        * /packet/queueing-block/out-pkts,
+        * /packet/queueing-block/out-bytes
+    *   Define variance = 0.98.
+    *   Start traffic.
+    *   Wait for 30 secs.
+    *   Capture the out-pkts and out-bytes leaf values (2).
+    *   Calculation of step 2 - step 1.
+        *  30 * pps = out-pkts.
+        *  30 * pps * 300 = out-bytes.
+        *  Capture tgen rx packets, tgen rx packets/step(2) values >= variance.
+    *   Stop traffic.
+
+
+*   Drop Counters - Lookup-Block
+
+    *   Subscribe to the following, and capture the leaf values(1)
+        * /drop/lookup-block/no-route
+        * /drop/lookup-block/no-nexthop
+        * /drop/lookup-block/invalid-packet
+        * /drop/lookup-block/forwarding-policy (TBD)
+        * /drop/lookup-block/incorrect-software-state
+        * /drop/lookup-block/rate-limit (TBD)
+        * /drop/lookup-block/acl-drops
+    
+    *   Start traffic, verify traffic received on atePort2.
+    *   Shutdown ateport2. Wait for 30 secs.
+    *   Verify /drop/lookup-block/no-route counter populated (2).
+        * Step 2-Step 1(no-route), calculation 30 * pps.       
+    *   Unshut ateport2, verify traffic recovers on atePort2.
+
+    *   Configure static Null0 route for outerDst_IP1.
+    *   After 30 secs, Verify /drop/lookup-block/no-nexthop leaf populated(3).
+        * Step 3- Step 1(no-nexthop), calcuation 30 * pps.
+    *   Unconfigure static route, and verify traffic recovers on atePort2, stop traffic.
+
+    *   Send traffic with custom checksum in ipv4 header.
+    *   After 30 secs, Verify /drop/lookup-block/invalid-packet leaf populated(4).
+        * Step 4 - Step 1(invalid-packet), calculation 30*pps.
+    *   Stop traffic. 
+
+    *   /drop/lookup-block/forwarding-policy <TBD>
+
+    *   Send traffic to Outer_Dst_IP1, and verify traffic received on atePort2.
+    *   Remove ipv4 address on dutport1.
+    *   After 30 secs, Verify /drop/lookup-block/incorrect-software-state populated(5).
```

Comment:
Actuallay, how DUT can judge that lack of IP address is incorrect? It follows controller intent, so IT IS CORRECT STATE fron DUT point of view.
To exercise this counter we would need to homehow introduce illegal register combination on ASIC. IMHO this is impossible to test in black-box acenario, outside of vendor engineering lab.

Reply:
Have verified this case, where /drop/lookup-block/incorrect-software-state gets populated with the mentioned trigger. Sending ipv4 traffic, with ip disabled on the interface causes this counter to populate due to incorrect state during lookup.